modules/launcher/src/main/java/org/mule/module/launcher/ParallelDeploymentDirectoryWatcher.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.launcher;
+
+import org.mule.module.launcher.application.Application;
+import org.mule.module.launcher.domain.Domain;
+import org.mule.module.launcher.util.ObservableList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Provides parallel deployment of Mule applications.
```

Comment:
What is version must be used on @since? This change has to be backported to some maintenance branches